### PR TITLE
merge RestProtocolTest.java

### PR DESCRIPTION
**Pull request by CrazyHZM**
## What is the purpose of the change

XXXXX

## Brief changelog

XXXXX

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by htynkn** [comment]
what's the purpose of this PR? add a new unit test?

**Comment by CrazyHZM** [comment]
> what's the purpose of this PR? add a new unit test?


Merged two RestProtocolTest.java and removed the extra test class

File: dubbo-rpc/dubbo-rpc-rest/src/test/java/org/apache/dubbo/rpc/protocol/rest/RestProtocolTest.java [hunk 1]
```diff
@@ -49,16 +50,55 @@ public void tearDown() {
...
+    public void testRestProtocol() {
+        URL url = URL.valueOf("rest://127.0.0.1:5342/rest/say?version=1.0.0");
+        DemoServiceImpl server = new DemoServiceImpl();
+        ProviderModel providerModel = new ProviderModel(url.getServiceKey(), server, DemoService.class);
+        ApplicationModel.initProviderModel(url.getServiceKey(), providerModel);
+
+        Exporter<DemoService> exporter = protocol.export(proxy.getInvoker(server, DemoService.class, url));
+        Invoker<DemoService> invoker = protocol.refer(DemoService.class, url);
+        Assertions.assertFalse(server.isCalled());
+
```
Comment (khanimteyaz) [inline dubbo-rpc/dubbo-rpc-rest/src/test/java/org/apache/dubbo/rpc/protocol/rest/RestProtocolTest.java]: would you care to split the UT in more logic unit e.g.

```
@Test
public void testServerIsCalled() {
   URL url = URL.valueOf("rest://127.0.0.1:5342/rest/say?version=1.0.0");
        DemoServiceImpl server = new DemoServiceImpl();
        ProviderModel providerModel = new ProviderModel(url.getServiceKey(), server, DemoService.class);
        ApplicationModel.initProviderModel(url.getServiceKey(), providerModel);

        Exporter<DemoService> exporter = protocol.export(proxy.getInvoker(server, DemoService.class, url));
        Invoker<DemoService> invoker = protocol.refer(DemoService.class, url);
}
```

```
@Test
public void testRestProtocolReturnResult() {
   URL url = URL.valueOf("rest://127.0.0.1:5342/rest/say?version=1.0.0");
        DemoServiceImpl server = new DemoServiceImpl();
        ProviderModel providerModel = new ProviderModel(url.getServiceKey(), server, DemoService.class);
        ApplicationModel.initProviderModel(url.getServiceKey(), providerModel);

        Exporter<DemoService> exporter = protocol.export(proxy.getInvoker(server, DemoService.class, url));
        Invoker<DemoService> invoker = protocol.refer(DemoService.class, url);
       DemoService client = proxy.getProxy(invoker);
        String result = client.sayHello("haha");
        Assertions.assertTrue(server.isCalled());
        Assertions.assertEquals("Hello, haha", result);
        invoker.destroy();
        exporter.unexport();
}
```

To me in the suggested way, if anything breaks, looking at UT we can get more closer to place where it has break. What do you say?

Comment (CrazyHZM) [reply dubbo-rpc/dubbo-rpc-rest/src/test/java/org/apache/dubbo/rpc/protocol/rest/RestProtocolTest.java]: I think the impact is not great. The testRestProtocol itself is to detect whether the Rest Protocol is successfully used.

**Comment by ralf0131** [comment]
Hi, next time please clearly describe your pull request by filling in the XXXX, which will give the reviewer a better understanding of what you are trying to do...

**Pull request merged**